sdk/core/Azure.Core/src/Pipeline/HttpMessageExtensions.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Core.Pipeline
+{
+    /// <summary>
+    /// Extension methods for <see cref="HttpMessage"/>.
+    /// </summary>
+    public static class HttpMessageExtensions
+    {
+        /// <summary>
+        /// Sets the package name and version portion of the UserAgent telemetry value.
+        /// Note: If <see cref="DiagnosticsOptions.IsTelemetryEnabled"/> is false, this value is never used.
+        /// </summary>
+        /// <param name="message">The <see cref="HttpMessage"/>.</param>
+        /// <param name="userAgentString">The <see cref="SetUserAgentString"/>.</param>
+        public static void SetUserAgentString(this HttpMessage message, UserAgentString userAgentString)
```

Comment:
It's a bit confusing that the type `UserAgentString` is not actually a string. Should we rename it to something else?

Reply:
Yeah, I agree but I'm not sure what would be better - perhaps `UserAgent` or `UserAgentValue` ?

Reply:
I think people refer to the concept as "UA string". UserAgent is a bit ambiguous because it's a name of the header. UserAgentValue is OK (thought I slightly prefer UserAgentString). I could go either way.

Reply:
UserAgent makes sense to me - you are setting the value for the UserAgent.